pancancer_evaluation/utilities/classify_utilities.py

```diff
@@ -819,6 +862,33 @@ def generate_param_grid(cv_results, fold_no=-1):
     return pd.DataFrame(np.array(results_grid).T, columns=columns)
 
 
+def shuffle_by_cancer_type(y_df, seed):
+    y_copy_df = y_df.copy()
+    with temp_seed(seed):
+        for cancer_type in y_copy_df.DISEASE.unique():
+            is_cancer_type = (y_copy_df.DISEASE == cancer_type)
+            y_copy_df.loc[is_cancer_type, 'status'] = (
+                np.random.permutation(y_copy_df.loc[is_cancer_type, 'status'].values)
+            )
```

Comment:
You can take or leave this suggestion; I initially thought `is_cancer_type` denoted whether the given disease was a cancer type. 
```suggestion
            cancer_type_indices = (y_copy_df.DISEASE == cancer_type)
            y_copy_df.loc[cancer_type_indices, 'status'] = (
                np.random.permutation(y_copy_df.loc[cancer_type_indices, 'status'].values)
            )
```